test_staticjinja.py

```diff
@@ -31,6 +31,8 @@ def site(template_path, build_path):
     template_path.join('template1.html').write('Test 1')
     template_path.join('template2.html').write('Test 2')
     template_path.mkdir('sub').join('template3.html').write('Test {{b}}')
+    template_path.mkdir('sub1').join('.ignored2.html').write('Ignored 2')
```

Comment:
I'm not sure, is this actually getting used anywhere? It seems like we create these two files and then forget about them, never checking that Site ever does the right thing. Either we should do some testing with them or never add them.

I would lean towards adding some better tests, but I'm not sure this PR is the place to do it.

Reply:
Totally agreed, this is PR is not the right place to do so.
But what I think this is being used indirectly in a test
```
def test_template_names(site):
    site.staticpaths = ["static_css", "static_js", "favicon.ico"]
    expected_templates = set(['template1.html',
                              'template2.html',
                              'sub/template3.html',
                              'template4.html'])
    assert set(site.template_names) == expected_templates
```
Since we are matching `site.template_names` with `expected_templates` which does
not contains any ignored or partial templates included those two as well.

So it good to have them and we can add some more test in another PR.

Reply:
Whoops, you're totally right, I missed `test_template_names()` when I reviewed in the GitHub web app. In fact, I would almost say that we should add
```
template_path.mkdir('.ignoreds').join('ignored3.html').write('Ignored 3')
template_path.mkdir('_partials').join('partial5.html').write('Partial 5')
```
What do you think? I think this can go in a later PR though, let's get your improvements in...